sql/core/src/test/scala/org/apache/spark/sql/execution/datasources/SchemaPruningSuite.scala

```diff
@@ -597,8 +597,7 @@ abstract class SchemaPruningSuite
             |  and e.employer.name = c.employer.company.name)
             |""".stripMargin)
         checkScan(query,
-          "struct<name:struct<first:string,middle:string,last:string>," +
-            "employer:struct<id:int,company:struct<name:string,address:string>>>",
+          "struct<name:struct<first:string>,employer:struct<company:struct<name:string>>>",
```

Comment:
Hmm, pruned schema is different in 3.1?

Reply:
Yes looks like starting from 3.2 it failed to prune some nested fields (for example in this case `name.middle` and `name.last` are not used)

Reply:
Hmm, okay, but I think it is unrelated to this change.